src/Core/File/LightData.hpp

```diff
@@ -43,54 +56,121 @@ class RA_CORE_API LightData : public AssetData {
             linear( l ),
             quadratic( q ) {}
     };
-
-    /// CONSTRUCTOR
-    LightData( const std::string& name = "", const LightType& type = UNKNOWN );
-
-    LightData( const LightData& data ) = default;
-
-    /// DESTRUCTOR
+    // TODO : allow to define other attenuation function such \max\left(0,1-\frac{\left|x\right|^2}{r^2}\right)^2
+
+    /**
+     * Light constructor.
+     * Must be calle explicitely with a name for the light and its type.
+     */
+    explicit LightData( const std::string& name = "", const LightType& type = UNKNOWN );
+
+    /**
+     * Copy constructor.
+     * As a Light a a union like object, no default copy constructor could be generated. This will take care of copying the good members of the union.
+     * @param data
+     */
+    LightData( const LightData& data );
+
+    /// Destructor
     ~LightData();
 
-    /// NAME
+    /**
+     * Extension of of the AssetData interface so that we can modify an abcet afeter ts creation.
+     * This has no impact on the coherence of the LightData object and could be used without restriction.
+     */
     inline void setName( const std::string& name );
 
-    /// TYPE
-    inline LightType getType() const;
-    inline void setType( const LightType& type );
 
-    /// FRAME
+    /**
+     * Acces to the local frame of the light.
+     * @return the local frame
+     */
     inline Core::Matrix4 getFrame() const;
+
+    /**
+     * Set the local frame of the light.
+     * @param frame the local frame
+     */
     inline void setFrame( const Core::Matrix4& frame );
 
-    /// DATA
-    /*
-    inline std::shared_ptr<Ra::Engine::Light> getLight() const;
-    inline void setLight( std::shared_ptr<Ra::Engine::Light> light );
+/**
+ * \defgroup HelperSetters Helper functions to set the various light parameters.
+ * @{
+ */
+    /**
+     * Construct a directional light.
+     * A directional light is only defined by its color and its lighting direction.
+     * No attenuation on directional.
+     */
```

Comment:
It might be useful to say here that this function changes the type of light.
By the way, why these methods are not static initialization functions ? They don't take into account the state of the instance, but just set a new light from a set of parameters.

Reply:
OK, I though the "Construct a 'type' light, in this method and following, was sufficiently clear. I will precise that these methods change the object so that it become an object of the type inferred from the overloaded method called.
And these method are not static due to inheritance (in coding style) of quite discutable conception choices and programming approach in the Old Radium. Yes, we need to refactor the LightData construction process but this will imply modifying much more files than this PR need to modify to just add the management of area light sources.
We can do this refactoring now (this will imply to modify also geometry, material and animation asset management that suffer from quite similar problems and all the systems that handle asset loading. Quite a huge work i think) or integrate it on the Radium-v2 roadmap as this future roadmap will contain a lot of such refactoring to simplify the writing of loaders/writers and, more generally, the input/ouput general process.